### PR TITLE
feat(manager/asdf): support `maven`

### PR DESCRIPTION
**Pull request by MPV**
## Changes

Add support for this ASDF plugin:
- https://github.com/halcyon/asdf-maven.git
   - ...being the one referred to at: https://github.com/asdf-vm/asdf-plugins/blob/master/plugins/maven

For upgrading versions of `maven`:
- https://github.com/apache/maven

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/asdf/extract.spec.ts [hunk 1]
```diff
@@ -344,6 +345,13 @@ dummy 1.2.3
             depName: 'lua',
             extractVersion: '^v(?<version>\\S+)',
           },
+          {
+            currentValue: '3.9.5',
+            datasource: 'github-releases',
```
Comment (viceice) [inline lib/modules/manager/asdf/extract.spec.ts:350]: use maven datasource, see maven wrapper manager for sample

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 2]
```diff
@@ -332,6 +332,14 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
       extractVersion: '^v(?<version>\\S+)',
     },
   },
+  maven: {
+    asdfPluginUrl: 'https://github.com/halcyon/asdf-maven.git',
+    config: {
+      datasource: GithubReleasesDatasource.id,
```
Comment (MPV) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:338]:
```suggestion
      datasource: MavenDatasource.id,
```

also needs an:
`import { MavenDatasource } from '../../datasource/maven';`

File: lib/modules/manager/asdf/extract.spec.ts (hunk 1 above)
Comment (MPV) [reply lib/modules/manager/asdf/extract.spec.ts:350]:
```suggestion
            datasource: 'maven',
```

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 3]
```diff
@@ -332,6 +332,14 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
...
+      packageName: 'apache/maven',
+      extractVersion: '^maven-(?<version>\\S+)',
```
Comment (MPV) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:340]:
```suggestion
      versioning: 'maven',
```

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 4]
```diff
@@ -332,6 +332,14 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
       extractVersion: '^v(?<version>\\S+)',
     },
   },
+  maven: {
+    asdfPluginUrl: 'https://github.com/halcyon/asdf-maven.git',
+    config: {
+      datasource: GithubReleasesDatasource.id,
+      packageName: 'apache/maven',
```
Comment (MPV) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:339]:
```suggestion
      packageName: 'org.apache.maven:apache-maven',
```

File: lib/modules/manager/asdf/extract.spec.ts [hunk 5]
```diff
@@ -344,6 +345,13 @@ dummy 1.2.3
...
+            packageName: 'apache/maven',
```
Comment (MPV) [inline lib/modules/manager/asdf/extract.spec.ts:351]:
```suggestion
            packageName: 'org.apache.maven:apache-maven',
```

File: lib/modules/manager/asdf/extract.spec.ts [hunk 6]
```diff
@@ -344,6 +345,13 @@ dummy 1.2.3
...
+            depName: 'maven',
+            extractVersion: '^maven-(?<version>\\S+)',
```
Comment (MPV) [inline lib/modules/manager/asdf/extract.spec.ts:353]:
```suggestion
            versioning: 'maven',
```

File: lib/modules/manager/asdf/extract.spec.ts (hunk 1 above)
Comment (MPV) [reply lib/modules/manager/asdf/extract.spec.ts:350]: I'll try running some tests, but I've drafted suggestions now on what I think you may be after.
Let me know if you see something else I haven't already.

**Comment by MPV** [comment]
Oops, seems I forgot about this one, and created a duplicate over in:
- https://github.com/renovatebot/renovate/pull/26802

**Review by rarkins — COMMENTED** [review]
Suggested changes look good

**Review by viceice — COMMENTED** [review]
please do that and fix conflicts

**Pull request closed without merging**